qiskit/pulse/builder.py

```diff
@@ -0,0 +1,1018 @@
+            lazy: If false the circuit will be transpiled and pulse scheduled
+                immediately. Otherwise, it will extend the active lazy circuit
+                as defined above.
+        """
+        try:
+            iter(qubits)
+        except TypeError:
+            qubits = (qubits,)
+
+        qc = circuit.QuantumCircuit(self.num_qubits)
+        qc.append(gate, qargs=qubits)
+        self.call_circuit(qc, lazy=lazy)
+
+
+def build(backend=None,
+          schedule: Optional[Schedule] = None,
+          default_alignment: str = 'left',
+          default_transpiler_settings: Optional[Dict[str, Any]] = None,
+          default_circuit_scheduler_settings: Optional[Dict[str, Any]] = None
+          ) -> ContextManager[Schedule]:
+    """
+    A context manager for the imperative pulse builder DSL.
+
+    Args:
+        backend (BaseBackend): a Qiskit backend. If not supplied certain builder
+            functionality will be unavailable.
+        schedule: a *mutable* pulse Schedule in which to build your
+            pulse program.
+        default_alignment: Default scheduling alignment for builder.
+            One of ``left``, ``right``, ``sequential`` or an alignment
+            contextmanager.
+        default_transpiler_settings: Default settings for the transpiler.
+        default_circuit_scheduler_settings: Default settings for the
+            circuit to pulse scheduler.
+
+    Returns:
+        A new builder context which has the active builder inititalized.
+    """
+    return _PulseBuilder(
+        backend=backend,
+        schedule=schedule,
+        default_alignment=default_alignment,
+        default_transpiler_settings=default_transpiler_settings,
+        default_circuit_scheduler_settings=default_circuit_scheduler_settings)
+
+
+# Builder Utilities ############################################################
+class NoActiveBuilder(exceptions.PulseError):
+    """Raised if no builder context is active."""
+
+
+def _active_builder() -> _PulseBuilder:
+    """Get the active builder in the active context.
+
+    Returns:
+        The active active builder in this context.
+
+    Raises:
+        NoActiveBuilder: If a pulse builder function is called outside of a
+            builder context.
+    """
+    try:
+        return BUILDER_CONTEXTVAR.get()
+    except LookupError:
+        raise NoActiveBuilder(
+            'A Pulse builder function was called outside of '
+            'a builder context. Try calling within a builder '
+            'context, eg., "with pulse.build() as schedule: ...".')
+
+
+def active_backend():
+    """Get the backend of the currently active context.
+
+    Returns:
+        BaseBackend: The active backend in the currently active builder context.
+
+    Raises:
+        exceptions.PulseError: If the builder does not have a backend set.
+    """
+    builder = _active_builder().backend
+    if builder is None:
+        raise  BackendNotSet(
```

Comment:
To fix pylint error - `qiskit/pulse/builder.py:524:14: E271 multiple spaces after keyword`
```suggestion
        raise BackendNotSet(
```